client/connector.go

```diff
@@ -527,6 +349,12 @@ func (u *UpsertBatchBuilderImpl) PrepareUpsertBatch(tableName string, columnName
 		if err = upsertBatchBuilder.AddColumnWithUpdateMode(columnID, dataType, updateModes[colIndex]); err != nil {
 			return nil, 0, err
 		}
+
+		if column.IsEnumColumn() {
+			if err = u.prepareEnumCases(tableName, columnName, colIndex, columnID, rows, abandonRows, column.CaseInsensitive, column.DisableAutoExpand); err != nil {
```

Comment:
this maybe an overhead for most no need to extend cases.

Reply:
this will compare with local cache first. only new cases will be extended